From 4f217aaf4bca500f13b4c560a6bbb3708c8e5dc2 Mon Sep 17 00:00:00 2001
From: Bhaskar Upadhaya <Bhaskar.Upadhaya@nxp.com>
Date: Fri, 1 Jul 2016 04:37:53 +0530
Subject: [PATCH 69/93] armv8:pfe:frdm Reset external PHYs before driver
 starts

- Disable CONFIG_RESET_PHY_R, it is resetting PHYs after driver started
- Call reset_phy() before driver initialization

Signed-off-by: Bhaskar Upadhaya <Bhaskar.Upadhaya@nxp.com>
Signed-off-by: Anji J <anji.jagarlmudi@freescale.com>
---
 board/freescale/ls1012afrdm/eth.c |    1 +
 include/configs/ls1012afrdm.h     |    1 -
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/board/freescale/ls1012afrdm/eth.c b/board/freescale/ls1012afrdm/eth.c
index 90ed66b..be17310 100644
--- a/board/freescale/ls1012afrdm/eth.c
+++ b/board/freescale/ls1012afrdm/eth.c
@@ -46,6 +46,7 @@ int board_eth_init(bd_t *bis)
 	struct mdio_info mac1_mdio_info;
 	struct ccsr_scfg *scfg = (struct ccsr_scfg *)CONFIG_SYS_FSL_SCFG_ADDR;
 
+	reset_phy();
 
 	/*TODO Following config should be done for all boards, where is the right place to put this */
 	out_be32(&scfg->pfeasbcr, in_be32(&scfg->pfeasbcr) | SCFG_PPFEASBCR_AWCACHE0);
diff --git a/include/configs/ls1012afrdm.h b/include/configs/ls1012afrdm.h
index 432317d..3de0cea 100644
--- a/include/configs/ls1012afrdm.h
+++ b/include/configs/ls1012afrdm.h
@@ -23,7 +23,6 @@
 #define EMAC2_PHY_ADDR          0x1
 #define CONFIG_PHYLIB
 #define CONFIG_PHY_REALTEK
-#define CONFIG_RESET_PHY_R
 #endif
 /*
 * USB
-- 
1.7.9.5

